hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java

```diff
@@ -1860,6 +1928,55 @@ FSPermissionChecker getPermissionChecker()
     }
   }
 
+  public INode getInode(INode inode) {
+    return inodeMap.get(inode);
+  }
+  public INode getInodeFromTempINodeMap(long id) {
+    LOG.debug("getInodeFromTempINodeMap: id={}, TempINodeMap.size={}",
+        id, inodeMapTemp.size());
+    INode inode = new INodeWithAdditionalFields(id, null,
+        new PermissionStatus("", "", new FsPermission((short) 0)), 0, 0) {
```

Comment:
If we define a PermissionStatus object as a static object, then it looks like we have to define that static object inside the class, instead of this function. Would that create some unnecessary confusion, given we need that object only for this function?

Happy to make it a static object if we can define it inside this function.